sample/SampleHost/Program.cs

```diff
@@ -1,53 +1,57 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
-using System;
 using Microsoft.Azure.WebJobs;
-using Microsoft.Azure.WebJobs.Logging;
 using Microsoft.Extensions.Logging;
+using Microsoft.Azure.WebJobs.Hosting;
+using Microsoft.Extensions.Hosting;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Extensions.Configuration;
+using System.Collections.Generic;
+using Microsoft.Extensions.Options;
 
 namespace SampleHost
 {
     class Program
     {
-        static void Main(string[] args)
+        public static async Task Main(string[] args)
         {
-            var config = new JobHostConfiguration();
-            config.Queues.VisibilityTimeout = TimeSpan.FromSeconds(15);
-            config.Queues.MaxDequeueCount = 3;
-            config.LoggerFactory = new LoggerFactory().AddConsole();
-
-            if (config.IsDevelopment)
+            var builder = new HostBuilder()
+                .UseEnvironment("Development")
+                .ConfigureWebJobsHost(o =>
+                {
+                    // Example setting options properties:
+                    // o.HostId = "testhostid";
+                })
+                // These can be toggled independently!
+                .AddWebJobsLogging() // Enables WebJobs v1 classic logging 
+                .AddStorageForRuntimeInternals() // enables WebJobs to run distributed, via a storage account to coordinate
+                .AddStorageBindings() // adds [Blob], etc bindings for Azure Storage. 
+                .AddApplicationInsights()
+                .ConfigureAppConfiguration(config =>
+                {
+                    // Adding command line as a configuration source
+                    config.AddCommandLine(args);
+                    config.AddInMemoryCollection(new Dictionary<string, string>()
+                    {
+                        // Configuration options set from configuration providers:
+                        { "HostId", "testhostidfromprovider" }
+                    });
+                })
+                .ConfigureLogging(b =>
+                {
+                    b.SetMinimumLevel(LogLevel.Debug);
+                    b.AddConsole();
```

Comment:
Something is wrong with console logging. Run this sample host and look at the console - a bunch of garbage/binary characters are spewed to the console